mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -73,10 +73,15 @@ class GENXDialectLLVMIRTranslationInterface
     llvm::LLVMContext &llvmContext = moduleTranslation.getLLVMContext();
     llvm::Function *llvmFunc = moduleTranslation.lookupFunction(func.getName());
 
+    // Set calling convention for kernel
+    if (attribute.getName() ==
+        GENX::GENXDialect::getKernelFuncAttrName()) {
+      llvmFunc->setCallingConv(llvm::CallingConv::SPIR_KERNEL);
+    }
+
     // Set max_work_group_size metadata.
     if (attribute.getName() ==
         GENX::GENXDialect::getMaxWorkGroupSizeAttrName()) {
-      llvmFunc->setCallingConv(llvm::CallingConv::SPIR_KERNEL);
```

Comment:
Remove also lines 102 and 120 given we decided to set the SPIRV_KERNEL calling convention here rather than directly in the consumer.